flutter/lib/src/method_channel_helper.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:meta/meta.dart';
+
+/// Makes sure no invalid data is sent over method channels.
+@internal
+class MethodChannelHelper {
+  static Map<String, dynamic>? normalizeMap(Map<String, dynamic>? data) {
+    if (data == null) {
+      return null;
+    }
+    final mapToReturn = <String, dynamic>{};
+    data.forEach((key, value) {
+      if (_isPrimitive(value)) {
+        mapToReturn[key] = value;
+      } else if (value is List<dynamic>) {
+        mapToReturn[key] = _normalizeList(value);
+      } else if (value is Map<String, dynamic>) {
+        mapToReturn[key] = normalizeMap(value);
+      } else if (value is Object) {
+        mapToReturn[key] = value.toString();
+      }
+    });
+    return mapToReturn;
+  }
+
+  static List<dynamic> _normalizeList(List<dynamic> data) {
+    final listToReturn = <dynamic>[];
+    for (var element in data) {
+      if (_isPrimitive(element)) {
+        listToReturn.add(element);
+      } else if (element is List<dynamic>) {
+        listToReturn.add(_normalizeList(element));
+      } else if (element is Map<String, dynamic>) {
+        listToReturn.add(normalizeMap(element));
+      } else if (element is Object) {
+        listToReturn.add(element.toString());
+      }
```

Comment:
```suggestion
      } else {
        listToReturn.add(element.toString());
      }
```

I guess we can omit the `Object` here.